test/deployframework/context.go

```diff
@@ -262,44 +264,140 @@ func (ctx *DeployerCtx) Setup(expectInstallErr bool) (*reportingframework.Report
+		return nil, fmt.Errorf("failed to wait for the %s deployment to report a successful upgrade: %v", reportingOperatorDeploymentName, err)
+	}
+
+	if purgeReportDataSources {
+		// in the case where we deleted the ReportDataSources in the ctx.Namespace,
+		// we need to verify that the metering-operator has progressed to reconciling
+		// these resources before running any of the reporting tests
+		err = WaitForReportDataSources(ctx.Logger, ctx.MeteringClient, ctx.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to wait for the ReportDataSources to exist")
+		}
+	}
+
+	initialDelay := 10 * time.Second
+	if ctx.RunDevSetup {
+		initialDelay = 1 * time.Second
+	}
+	ctx.Logger.Infof("Upgrading metering took %v", time.Since(start))
+
+	routeBearerToken, err := GetServiceAccountToken(
+		ctx.Client,
+		initialDelay,
+		waitingForServiceAccountTimePeriod,
+		ctx.Namespace,
+		reportingOperatorServiceAccountName,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the route bearer token: %v", err)
+	}
+
+	reportResultsPath := filepath.Join(ctx.TestCaseOutputPath, reportResultsDir)
+	err = os.Mkdir(reportResultsPath, 0777)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the report results directory %s: %v", reportResultsPath, err)
+	}
+	ctx.Logger.Infof("Report results directory: %s", reportResultsPath)
+
+	rf, err := reportingframework.New(
+		useHTTPSAPI,
+		useKubeProxyForReportingAPI,
+		useRouteForReportingAPI,
+		ctx.Namespace,
+		routeBearerToken,
+		reportingAPIURL,
+		reportResultsPath,
+		ctx.Config,
+		ctx.Client,
+		ctx.MeteringClient,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct a reportingframework: %v", err)
+	}
+
+	return rf, nil
+}
+
 // Teardown is a method that creates the resource and container logging
 // directories, then populates those directories by executing the
 // cleanup bash script, while streaming the script output
 // to stdout. Once the cleanup script has finished execution, we can
 // uninstall the metering stack and return an error if there is any.
-func (ctx *DeployerCtx) Teardown() error {
-	var errArr []string
-
-	logger := ctx.Logger.WithFields(logrus.Fields{"component": "cleanup"})
-
+func (ctx *DeployerCtx) Teardown(uninstallFunc func() error) error {
+	var (
+		errArr []string
+		err    error
+	)
 	if ctx.RunTestLocal && ctx.LocalCtx != nil {
-		err := ctx.LocalCtx.CleanupLocal()
+		err = ctx.LocalCtx.CleanupLocal()
 		if err != nil {
 			errArr = append(errArr, fmt.Sprintf("failed to successfully cleanup local resources: %v", err))
 		}
 	}
 
+	err = ctx.MustGatherMeteringResources(cleanupScriptName)
+	if err != nil {
+		errArr = append(errArr, fmt.Sprintf("failed to successfully gather all of the metering resources: %v", err))
+	}
+
 	// Check if the user wants to run the E2E suite using the developer setup.
 	// If true, we skip the process of deleting and logging of the metering
 	// resources that were provisioned during the manual install.
```

Comment:
Is this comment block still valid and needed (Teardown has its own comment block) given the changes below?

Reply:
That needs to be updated.